### PR TITLE
Add command line argument checks

### PR DESCRIPTION
**Pull request by blackHatMonkey**
Add the function to check if an interface supports a specific mode. This is part of a multiple PR that essentially allows us to check if the provided interface in the command line arguments is valid. So an example of this would be:
```
wifiphisher --ap-interface HAHA
usage: wifiphisher.py ...
error: argument --ap-interface: HAHA is not a valid interface
```
This allows us to use this interface right away in the program since we know its valid.

Also I placed this function in the `constants` since this function should also be used when finding interfaces.

### CODE REVIEW

File: wifiphisher/common/constants.py [hunk 1]
```diff
@@ -132,3 +134,24 @@
 
 # Fourway handshake extension
 CONST_A = "Pairwise key expansion"
+
+
+def does_have_mode(interface, mode):
```
Comment (sophron) [inline wifiphisher/common/constants.py]: `constants.py` is only for constants. What does this method has to do here?

Comment (blackHatMonkey) [reply wifiphisher/common/constants.py]: >Also I placed this function in the constants since this function should also be used when finding interfaces.

As I have said earlier this function is used in multiple modules. Therefore it is necessary for it to be in the `constants`. It didn't make sense to me to create another file which needs to be shard between multiple modules since we already have `constants`. That being said the name can be changed in the future to reflect the content better.

Comment (sophron) [reply wifiphisher/common/constants.py]: > It didn't make sense to me to create another file which needs to be shard between multiple modules since we already have constants. 

But that's the right way to do it (logic separation). Why do you think your method doesn't fit under the `interfaces.py` file?

Comment (blackHatMonkey) [reply wifiphisher/common/constants.py]: This is because I was going to create a `arguments` module where we do all the argument handling and checking. It doesn't make sense to import interfaces in there does it?

Comment (sophron) [reply wifiphisher/common/constants.py]: Of'course it does. If your module will do argument checking, then it will involve many interface-related functions and it makes sense to import `interfaces.py`.

However, we currently have `opmode.py`. Make sure there are distinct roles between `opmode.py` and your module.

Comment (blackHatMonkey) [reply wifiphisher/common/constants.py]: It looks similar to what `_check_args` does except it will only check for that specific argument (it doesn't have access to all of the arguments given. So perhaps some of the stuff in `_check_args` can be removed after this.

The general prototype for the functions are like so:
```python
def validate_ap_interface(interface):
    if interface is not valid:
        raise argparse.error("interface is invalid")

```

Comment (sophron) [reply wifiphisher/common/constants.py]: You need to be a little more descriptive. What will this module check exactly? Will it just check if the interfaces are right?  How exactly does it differ from `opmode.py`? Why do we need a separate module and we don't extend `opmode.py` instead?

Comment (blackHatMonkey) [reply wifiphisher/common/constants.py]: This module does all the validation for the arguments that the user has entered. As I said earlier they share some common code but that's not all. It has an advantage of only getting executed when necessary so that means that we don't have do the check wether the user has inputted a certain option or not. Another thing that is nice about this approach is that it shows a nice error before even starting the program so it is clear that there is a error in the user input.

So to summarize the module will have a method for each argument that needs to be validated.

Comment (sophron) [reply wifiphisher/common/constants.py]: > As I said earlier they share some common code but that's not all.

If they share common code, why don't we extend `opmode.py` instead?

>  It has an advantage of only getting executed when necessary so that means that we don't have do the check wether the user has inputted a certain option or not.

I'm not sure I understand this advantage.

> Another thing that is nice about this approach is that it shows a nice error before even starting the program so it is clear that there is a error in the user input.

Isn't this the same as `check_args` in `opmode.py`?

I don't know if I have understood exactly what you are proposing but it sounds like it could be included in the `opmode` file.

Comment (blackHatMonkey) [reply wifiphisher/common/constants.py]: I don't mind putting in the `opmode` module at all.
>I'm not sure I understand this advantage.

Take a look at the following example:
https://github.com/wifiphisher/wifiphisher/blob/185f9ae8abbb86918db7246650a2cf0041419046/wifiphisher/common/opmode.py#L64-L73

This can be changed to 
```python
def validate_password(password):
    length = len(password)
    
    if length < 8 or length > 64:
        raise argparse.error("Length of the password is wrong")

```

We can easily test this function and all we need to do is add a type to the argument
```python
parser.add_argument(
        "-pK",
        "--presharedkey",
        type=opmode.validate_password,
        help=("Add WPA/WPA2 protection on the rogue Access Point. " +
"Example: -pK s3cr3tp4ssw0rd"))
```

As a bonus you can try running the following example to see the output:
```python
import argparse


parser = argparse.ArgumentParser(description='Process some integers.')
parser.add_argument('--integers', type=c.tan, help='an integer for the accumulator')
parser.add_argument("--done", type=str, help="just a string bro")

args = parser.parse_args()

```

**Comment by sophron** [comment]
Got it now.

Yes, let's extend `opmode.py` now for this.

**Comment by blackHatMonkey** [comment]
@sophron This one is done and I have done the integration testing for it. However I found something weird happening even on master. Does providing both the `-aI` and `-jI` arguments work for you? I keep getting the error that these two should be used in conjunction. After bit of testing it seems that `-aI` is never set so it holds `None` which activates the check in the `opmode`.

**Comment by sophron** [comment]
@blackHatMonkey I'll merge this soon. 

No, I'm not aware of the issue that you describe. I'll have a look.

**Comment by sophron** [comment]
Good, I also added one more check to make sure that the given interface is wireless.
